### PR TITLE
Improve `configurable` BBE

### PR DESCRIPTION
**Pull request by HindujaB**
## Purpose
 Include information on record & table type support

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

## Related PRs
https://github.com/ballerina-platform/ballerina-lang/pull/28461

### CODE REVIEW

File: examples/configurable/configurable.bal [hunk 1]
```diff
@@ -18,10 +25,17 @@ configurable float maxPayload = 1.0;
 // provided for it in the `Config.toml` file.
 configurable string protocol = "http";
 
+// A `configurable` variable `admin` with record type `UserInfo & readonly` is initialized
```
Comment (praneesha) [inline examples/configurable/configurable.bal:28]:
```suggestion
// A `configurable` variable named `admin` of the `UserInfo & readonly` record type is initialized.
```

File: examples/configurable/configurable.bal [hunk 2]
```diff
@@ -18,10 +25,17 @@ configurable float maxPayload = 1.0;
...
+configurable UserInfo & readonly admin = ?;
+// A `configurable` variable `users` with table type `table<(UserInfo & readonly)> key(username)` is initialized
```
Comment (praneesha) [inline examples/configurable/configurable.bal:30]:
```suggestion
// A `configurable` variable named `users` of the `table<(UserInfo & readonly)> key(username)` table type is initialized.
```

File: examples/configurable/configurable.description [hunk 3]
```diff
@@ -10,4 +10,5 @@
 //
 // The configuration file must be in the TOML(v0.4) format.
 // Ballerina currently supports configurable variables of types `int`, `float`,
-// `boolean`, `string`, `decimal` and the arrays of respective types.
+// `boolean`, `string`, `decimal`, the arrays of the respective types, record and
```
Comment (praneesha) [inline examples/configurable/configurable.description:13]:
```suggestion
// `boolean`, `string`, `decimal`, the arrays of the respective types, record, and
```

**Comment by praneesha** [comment]
@HindujaB - Shall we include the content for [1] also in this PR?

[1] https://github.com/ballerina-platform/ballerina-lang/issues/28741

**Comment by HindujaB** [comment]
> @HindujaB - Shall we include the content for [1] also in this PR?
> 
> [1] [ballerina-platform/ballerina-lang#28741](https://github.com/ballerina-platform/ballerina-lang/issues/28741)

We prefer to keep the BBE simple as possible. Therefore the BBE has only single bal file execution scenario and not mentions about package level execution. The information mentioned in the issue is related to package execution. 

And we are having a plan to include a separate learn page for configurables. This information can be included there.

**Comment by praneesha** [comment]
> > @HindujaB - Shall we include the content for [1] also in this PR?
> > [1] [ballerina-platform/ballerina-lang#28741](https://github.com/ballerina-platform/ballerina-lang/issues/28741)
> 
> We prefer to keep the BBE simple as possible. Therefore the BBE has only single bal file execution scenario and not mentions about package level execution. The information mentioned in the issue is related to package execution.
> 
> And we are having a plan to include a separate learn page for configurables. This information can be included there.


+1

**Review by praneesha — APPROVED** [review]

**Pull request merged**